lib/sender.js

```diff
@@ -27,6 +27,20 @@ class Sender {
     this._bufferedBytes = 0;
     this._deflating = false;
     this._queue = [];
+
+    this._socket.on('close', () => {
+      const err = new Error(
+        `WebSocket is not open: readyState ${this._socket.readyState} `
```

Comment:
Good point.  Would it be better to have a simple generic message here?  The original intent was to provide consistency with what the user would get as an error regardless of whether their send attempt was queued or not, but since it was already inconsistent a generic message wouldn't break anything here.

Reply:
Maybe we can pass the `Sender` a reference of the `WebSocket` so we can:

1. Set the ready state to `CLOSING`
2. Use it to make error message consistent.

What do you think?

Reply:
I've been trying to avoid that because I don't like how it hard links Sender to WebSocket.  A Sender doesn't need a WebSocket to function.

Reply:
I agree, the same is also true for the `Receiver` but we add a WebSocket reference to it in order to use it in the event listeners.